src/components/Visualization/Visualization.js

```diff
@@ -108,24 +108,50 @@ export const Visualization = ({
     displayProperty,
     onResponsesReceived,
     onColumnHeaderClick,
+    onDataSorted,
     onError,
 }) => {
     const [uniqueLegendSets, setUniqueLegendSets] = useState([])
-    const [{ sortField, sortDirection, pageSize, page }, setSorting] =
-        useReducer((sorting, newSorting) => ({ ...sorting, ...newSorting }), {
-            sortField: null,
-            sortDirection: DEFAULT_SORT_DIRECTION,
-            page: FIRST_PAGE,
-            pageSize: PAGE_SIZE,
-        })
 
     const visualization = useMemo(() => AO && transformVisualization(AO), [AO])
 
     const visualizationRef = useRef(visualization)
 
+    const getSorting = (visualization) => {
+        if (
+            visualization &&
+            visualization.sorting &&
+            visualization.sorting.length
```

Comment:
refactor to `if ( visualization?.sorting?.length )`?

Reply:
or even just skip the `if` and do a conditional return like `cost getSorting = (visualization) => visualization?.sorting?.length ? { foo } : { bar }`

Reply:
Refactored.